include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -2429,6 +2429,60 @@ DataOut_DoFData<dim, patch_dim, spacedim, patch_spacedim>::
             Assert(false, ExcNotImplemented());
         }
 
+  // Automatically merge components which are located consequently and have the
+  // same name into a vector or a tensor.
+  if (patch_spacedim > 1)
+    {
+      const auto names = get_dataset_names();
+
+      for (unsigned int component_start = 0; component_start < names.size();)
+        {
+          DataComponentInterpretation::DataComponentInterpretation
+            component_interpretation =
+              DataComponentInterpretation::component_is_scalar;
+
+          unsigned int component_end = component_start + 1;
+          for (; component_end < names.size(); ++component_end)
+            if (names[component_end] != names[component_start])
+              break;
+
+          const unsigned int n_components = component_end - component_start;
+
+          if (n_components == patch_spacedim)
+            component_interpretation =
+              DataComponentInterpretation::component_is_part_of_vector;
+          else if (n_components == patch_spacedim * patch_spacedim)
+            component_interpretation =
+              DataComponentInterpretation::component_is_part_of_tensor;
+
+          if (component_interpretation !=
+              DataComponentInterpretation::component_is_scalar)
+            {
+              bool add_new_range = true;
+              for (const auto &range : ranges)
+                {
+                  // Check if the component is already in the ranges
+                  if (std::get<0>(range) == component_start)
+                    add_new_range = false;
+
+                  // Here we rely on the fact that the ranges are sorted in
+                  // accending order
+                  if (std::get<0>(range) > component_start)
+                    break;
```

Comment:
`ranges` is not empty at the beginning?

Reply:
There already can be something there coming from this loop https://github.com/dealii/dealii/blob/4ec36db9eed1e46fb96b5787af7e80d9ab45b180/include/deal.II/numerics/data_out_dof_data.templates.h#L2263-L2430